compiler/rustc_ast/src/ast.rs

```diff
@@ -335,14 +335,15 @@ pub enum ParamKindOrd {
     // is active. Specifically, if it's only `min_const_generics`, it will still require
     // ordering consts after types.
     Const { unordered: bool },
+    Infer,
```

Comment:
Introducing `Infer` at AST level should be unnecessary, it doesn't make any difference before lowering to HIR and can be introduced during it.

Reply:
This type is just for convenience with `Ord`, and I believe it is used for sorting the params during HIR lowering. My understanding may be incorrect, but this was my intended purpose for adding it here.